assets/section-main-product.css

```diff
@@ -1142,6 +1142,38 @@ a.product__text {
   border-color: rgb(var(--color-foreground));
 }
 
+.image--full-size {
+  cursor: zoom-out;
+  z-index: 1;
+  margin: 0;
+}
+
+.image--hover {
```

Comment:
Could we consider naming the classes related to the zoom functionality slightly less generically? 
Something like `.image-magnify` and `.image-magnify__full-size` or `.image-hover-zoom` and `.image-hover-zoom__full-size`. I mention the "magnify" term because matching the target classnames to the js terminology could make the relationship a little clearer.

I also don't think we have a base `.image` class being used so the `--full-size` and `--hover` modifiers don't align exactly with our intended (but probably inconsistent) naming practices.